### PR TITLE
Cart access

### LINKED ISSUE #43: User Story 30, Visitors must register or log in to check out
```
As a visitor
When I have items in my cart
And I visit my cart
I see information telling me I must register or log in to finish the checkout process
The word "register" is a link to the registration page
The words "log in" is a link to the login page
```

### Checklist:

- [ ] base tests are written
- [ ] base code is written to pass tests
- [ ] edge case ("sad path") tests written (if required)
- [ ] edge case code written (if required)
- [ ] code is reviewed
- [ ] branch is merged
- [ ] all tests still pass

child of #49

child of #79

depends on #47

depends on #76

depends on #70

depends on #66

### PR DESCRIPTION
**Pull request by teresa-m-knowles**
As a visitor
When I have items in my cart
And I visit my cart
I see information telling me I must register or log in to finish the checkout process
The word "register" is a link to the registration page
The words "log in" is a link to the login page

Closes #43

screen shot 2019-02-16 at 9 31 07 pm

### CODE REVIEW

File: app/controllers/carts_controller.rb [hunk 1]
```diff
@@ -4,8 +4,13 @@ class CartsController < ApplicationController
   before_action :require_shopper
 
   def show
-    flash[:primary] = "Your cart is empty." if @cart.total_count == 0
     set_cart
+    flash[:primary] = "Your cart is empty." if @cart.total_count == 0
+
+    if !current_user && @cart.total_count > 0
+
+      flash[:danger] = %Q[You must <a href="/login">login</a> or <a href="/register">register</a> to checkout.].html_safe
```
Comment (teresa-m-knowles) [inline app/controllers/carts_controller.rb:12]: %Q[Hello] is the same as "Hello". Just added that to then be able to add the double quotes in the link to make it clearer.

html_safe is because, without it, the view will literally print that exact string, it escapes the html.

**Review by allieIsOkay — APPROVED** [review]
Looks great!

File: app/controllers/carts_controller.rb (hunk 1 above)
Comment (timnallen) [reply app/controllers/carts_controller.rb:12]: Nice work!

**Review by csvlewis — APPROVED** [review]
This all looks good! I'll merge it now.
